libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -606,7 +606,8 @@ public String getNewAuthToken() {
                         	if (Looper.myLooper() == null) {
                                 Looper.prepare();
                         	}
-                            SalesforceSDKManager.getInstance().logout(null, false);
+                            SalesforceSDKManager.getInstance()
+                                    .logout(null, null, false, OAuth2.LogoutReason.TOKEN_EXPIRED);
```

Comment:
When the refresh token has expired, there won't be a (successful) client call to the server to revoke the token and therefore the logout reason will never be sent to the server. But the server should have created a log line when the refresh token expires.

TLDR: TOKEN_EXPIRED reason can never be logged by the client only by the server

Reply:
That being said, it's fine to use that reason here (even though it won't leave the client).

Reply:
Great callout.  I think I will leave it as there are no negative side affects but I will add a comment noting this.

Reply:
I'll update on iOS too